src/content/advanced/gitops/apps/add_app_template.md

```diff
@@ -0,0 +1,99 @@
+---
+linkTitle: Add a new App Template
+title: Add a new App Template
+description: Learn how to create application templates to deploy apps using GitOps.
+weight: 50
+menu:
+  main:
+    identifier: advanced-gitops-apps-add-app-template
+    parent: advanced-gitops-apps
+last_review_date: 2023-02-10
+user_questions:
+  - How can I create an template for app deployment in GitOps?
+owner:
+  - https://github.com/orgs/giantswarm/teams/team-honeybadger
+---
+
+# Add a new App Template to the repository
+
+In order to avoid adding the same application from scratch across all your clusters, you can prepare App Templates providing a pre-configured version of an App. This also allows you to manage and version an app's configuration even if the app itself is not yet installed in any cluster.
+
+## Example
+
+An example of an App Template is available in the [gitops-template repository "bases/apps/nginx-ingress-controller"](https://github.com/giantswarm/gitops-template/tree/main/bases/apps/nginx-ingress-controller).
+
+## Export environment variables
+
+__Note__: Management Cluster codename, Organization name, Workload Cluster name and several App-related values are needed in multiple places across this instructions, the least error prone way of providing them is by exporting them as environment variables:
+
+```nohighlight
+export WC_NAME=CLUSTER_NAME
+export APP_NAME="\${cluster_name}-APP_NAME"
+export APP_VERSION=APP_VERSION
+export APP_CATALOG=APP_CATALOG
+export APP_NAMESPACE=APP_NAMESPACE
+# OPTIONAL
+export APP_USER_VALUES=CONFIGMAP_OR_SECRET_PATH
+```
+
+## Setting up directory tree structure for managing apps
+
+1. Go to the `apps` directory and prepare a directory for the new App Template:
+
+    ```nohighlight
+    cd bases/apps/
+    mkdir ${APP_NAME}
+    ```
+
+1. Go to the newly created directory and use [the kubectl-gs plugin](https://github.com/giantswarm/kubectl-gs) to generate the [App CR](https://docs.giantswarm.io/ui-api/kubectl-gs/template-app/):
+
+    ```nohighlight
+    cd ${APP_NAME}/
+    kubectl gs template app \
+    --app-name ${APP_NAME} \
+    --catalog ${APP_CATALOG} \
+    --cluster ${WC_NAME} \
+    --name ${APP_NAME} \
+    --namespace ${APP_NAMESPACE} \
+    --version ${APP_VERSION} > appcr.yaml
+    ```
+
+__Note__: you most likely want to provide a default configuration, so add any of the below flags to the previous command:
+
+    ```nohighlight
+    --user-configmap ${APP_USER_VALUES}
+    --user-secret ${APP_USER_VALUES}
+    ```
+
+**__Note__**, We're including `${cluster_name}` in the app name to avoid a problem when two or more clusters in the same organization want to deploy the same app with its default name.
```

Comment:
```suggestion
__Note__: We're including `${cluster_name}` in the app name to avoid a problem when two or more clusters in the same organization want to deploy the same app with its default name.
```